public/static/app/perf-bb/PerfBBRejectsCtrl.js

```diff
@@ -0,0 +1,385 @@
+    },
+  ];
+
+  vm.state = new RejectState(project, vm, $scope);
+
+  // Creates aggregation expression, which could be used by Stitch
+  // for given `state`
+  $scope.getAggChain = (state) => {
+    let chain = [];
+
+    // Check if the state has filters
+    let filteringChain = MDBQueryAdaptor.compileFiltering(
+      // filters context enhances state data with important meta data
+      state.getFilteringContext()
+    );
+
+    if(!filteringChain) {
+      filteringChain = {$match:{}};
+    }
+    if(!filteringChain['$match']) {
+      filteringChain['$match'] = {};
+    }
+    filteringChain['$match']['$or'] = [{"rejected": true}, {"results.rejected": true}];
+    chain.push(filteringChain);
+
+    const sortingChain = MDBQueryAdaptor.compileSorting(state.sorting);
+    // check if sorting query was compiled into something
+    if (sortingChain) {
+      chain.push(sortingChain);
+    }
+
+    chain.push({$limit: LIMIT});
+    return chain;
+  };
+
+  $scope.sortRevision = (a, b, rowA, rowB) => {
+    // Sort revision by order instead of revision id.
+    const nulls = vm.gridApi.core.sortHandleNulls(a, b);
+    if (nulls !== null) {
+      return nulls;
+    }
+
+    if (a === b) {
+      return 0;
+    }
+
+    if (rowA && rowB) {
+      return rowA.entity.order - rowB.entity.order;
+    }
+
+    return a - b;
+  };
+
+  function hydrateData(docs, whitelist) {
+    return _.each(docs, (doc) => {
+      const matcher = _.pick(doc, 'project', 'variant', 'task', 'revision');
+      doc._buildId = EvgUtil.generateBuildId({
+        project: project,
+        revision: doc.revision,
+        buildVariant: doc.variant,
+        createTime: doc.create_time,
+      });
+      doc.whitelisted = (whitelist ? _.findWhere(whitelist, matcher) !== undefined : false);
+    });
+  }
+
+  // Required by loadData.
+  let theMostRecentPromise;
+
+  function setInitialGridFiltering(state) {
+    _.each(state.filters, function (term, colName) {
+      const col = $scope.getCol(colName);
+      if (!col) return;  // Error! Associated col does not found
+      col.filters = [{term: term}];
+    });
+  }
+
+  // Sets `state` to grid filters
+  function setInitialGridState(state) {
+    setInitialGridFiltering(state);
+
+    _.each(state.sorting, function (sortingItem) {
+      const col = $scope.getCol(sortingItem.field);
+      if (!col) return; // Error! Associated col does not found
+      col.sort.direction = sortingItem.direction;
+    });
+  }
+
+  // Load data:
+  //    starts the loading indicator
+  //    get the remote state from atlas
+  //    renders the data (if it is the latest data load event)
+  //    clears the loading indicator.
+  // Setting filters and updating other vm state is done through relaod() and most methods should
+  // use reload..
+  function loadData() {
+    vm.isLoading = true;
+    vm.gridOptions.data = [];
+    const promises = {
+      docs:Stitch.use(STITCH_CONFIG.PERF).query((db) => {
```

Comment:
[nit] Could we put a space between the ':' and the values in this object?

Reply:
done